lib/translate/status_translator.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIB_STATUS_TRANSLATOR_H
+#define LIB_STATUS_TRANSLATOR_H
+
+#include "../../third_party/spdk_defs/nvme_defs.h"
+#include "../scsi_defs.h"
+#include "../util/optional.h"
+
+namespace translator {
+
+struct TranslatedScsiStatus {
+  scsi_defs::Status status;
+  scsi_defs::SenseKey sense_key;
+  util::Optional<scsi_defs::AdditionalSenseCode> sense_code;
+
+  TranslatedScsiStatus() {}
+  TranslatedScsiStatus(
+      scsi_defs::Status status, scsi_defs::SenseKey sense_key,
+      util::Optional<scsi_defs::AdditionalSenseCode> sense_code)
+      : status{status}, sense_key{sense_key}, sense_code{sense_code} {}
+};
+
+util::Optional<TranslatedScsiStatus> translateStatusCode(uint8_t sct,
+                                                         uint8_t sc);
+util::Optional<TranslatedScsiStatus> translateStatusCode(
+    nvme_defs::GenericCommandStatusCode sc);
+util::Optional<TranslatedScsiStatus> translateStatusCode(
+    nvme_defs::CommandSpecificStatusCode sc);
+util::Optional<TranslatedScsiStatus> translateStatusCode(
+    nvme_defs::MediaErrorStatusCode sc);
```

Comment:
[Functions should be named in PascalCase](https://google.github.io/styleguide/cppguide.html#Function_Names).

Reply:
I'll try to be more careful with styling in the future!